tests/framework/resourcemanager.go

```diff
@@ -71,7 +74,17 @@ func (rm *ResourceManager) Apply(resources []client.Object) error {
 	defer cancel()
 
 	for _, resource := range resources {
-		if err := rm.K8sClient.Get(ctx, client.ObjectKeyFromObject(resource), resource); err != nil {
+		var obj client.Object
+
+		unstructuredObj, ok := resource.(*unstructured.Unstructured)
+		if ok {
+			obj = unstructuredObj.DeepCopy()
+		} else {
+			t := reflect.TypeOf(resource).Elem()
+			obj = reflect.New(t).Interface().(client.Object)
+		}
+
+		if err := rm.K8sClient.Get(ctx, client.ObjectKeyFromObject(resource), obj); err != nil {
```

Comment:
the existing code
https://github.com/nginxinc/nginx-gateway-fabric/blob/4d5c663c6469ea6a5507492d29cd561ce2432d82/tests/framework/resourcemanager.go#L73-L89
will never update the object, because it will get the resource into the `resource` variable that contains an update, thus overwriting the update to the current state. As a result, `rm.K8sClient.Update(ctx, resource);` will update the resource to its current state.

Reply:
Ah, nice